cli/Cargo.toml

```diff
@@ -25,9 +25,29 @@ is-it-maintained-open-issues = { repository = "https://github.com/hyperledger/ir
 maintenance = { status = "actively-developed" }
 
 [dependencies]
-iroha_core = { version = "=2.0.0-pre-rc.2", path = "../core" }
+iroha_core  = { version = "=2.0.0-pre-rc.2", path = "../core" }
 iroha_permissions_validators = { version = "=2.0.0-pre-rc.2", path = "../permissions_validators" }
 iroha_logger = { version = "=2.0.0-pre-rc.2", path = "../logger"}
+iroha_futures = { version = "=2.0.0-pre-rc.2", path = "../futures" }
+iroha_actor = { version = "=2.0.0-pre-rc.2", path = "../actor" }
+iroha_data_model = { version = "=2.0.0-pre-rc.2", path = "../data_model" }
+iroha_telemetry = { version = "=2.0.0-pre-rc.2", path = "../telemetry", optional = true }
+iroha_version = { version = "=2.0.0-pre-rc.2", path = "../version", features = ["warp"] }
+iroha_config = { version = "=2.0.0-pre-rc.2", path = "../config" }
+iroha_crypto = { version = "=2.0.0-pre-rc.2", path = "../crypto" }
+iroha_p2p = { version = "=2.0.0-pre-rc.2", path = "../p2p" }
 
+eyre = "0.6.5"
+futures = { version = "0.3.17", default-features = false, features = ["std", "async-await"] }
+parity-scale-codec = { version = "2.3.1", default-features = false, features = ["derive"] }
+serde = { version = "1.0", features = ["derive"] }
+serde_json = "1.0"
+tokio = { version = "1.6.0", features = ["sync", "time", "rt", "io-util", "rt-multi-thread", "macros", "fs"]}
+warp = "0.3"
+thiserror = "1.0.28"
 color-eyre = "0.5.11"
-tokio = { version = "1.6.0", features = ["rt", "rt-multi-thread"]}
+
+[dev-dependencies]
+unique_port = "0.1.0"
+iroha_core = { version = "=2.0.0-pre-rc.2", path = "../core", features = ["cross_crate_testing"] }
```

Comment:
That's a bug in the dependency resolution, and AFAIK it will be fixed eventually. For now, I think that marking everything that isn't meant for release builds as optional, even if it gets compiled is still an improvement.

Reply:
One way to mitigate this is to extract code required in tests which needs to be shared among more crates into a separate crate

Reply:
also: https://doc.rust-lang.org/cargo/reference/resolver.html#feature-resolver-version-2

Reply:
Resolver = 2 seems to be what we need.